### ISSUE TITLE
Incorrect `is_empty` result after intersection

### ISSUE THREAD

**Issue by 3Hren** [tower120/hi_sparse_bitset#33]
Hi! First or all, thanks for this magnificiant library!

I have the following code:
```rust
let mut bm0: BitSet<_64bit> = BitSet::new();
bm0.insert(0);
bm0.insert(1);
bm0.insert(512);
bm0.insert(800);

let mut bm1: BitSet<_64bit> = BitSet::new();
bm1.insert(1);
bm1.insert(2);
bm1.insert(511);
bm1.insert(513);
bm1.insert(800);

let intersection = &bm0 & &bm1;
dbg!(&intersection);
assert!(!intersection.is_empty());
```

, which fails in assertion, while the `dbg` macro shows that the intersected bitset is not empty.

Am I doing something wrong?

**Comment by tower120** [comment]
There is an error in !`TRUSTED_HIERARCHY`'s `is_empty()` implementation.
I'll fix that soon.

Thank you for bug report.

**Comment by tower120** [comment]
Can you try this branch https://github.com/tower120/hi_sparse_bitset/tree/incorrect_is_empty_fix ?

**Comment by tower120** [comment]
fixed in 0.6.1

**Comment by 3Hren** [comment]
> Can you try this branch https://github.com/tower120/hi_sparse_bitset/tree/incorrect_is_empty_fix ?

Now everything works. Thank you!
